pkg/api/injectsnippetsfs.go

```diff
@@ -0,0 +1,103 @@
+package api
+
+import (
+	"bytes"
+	"fmt"
+	"html"
+	"io"
+	"io/fs"
+	"time"
+)
+
+type InjectSnippetsFS struct {
+	fs.FS
+	name    string
+	content string
+}
+
+func NewInjectIndexFS(fsys fs.FS, name string, marker string, snippets map[string]string) (fs.FS, error) {
+	if len(snippets) == 0 {
+		// no snippets, return the original marker
+		return fsys, nil
+	}
+
+	// read the content and inject snippets
+	f, err := fsys.Open(name)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { _ = f.Close() }()
+	all, err := io.ReadAll(f)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", name, err)
+	}
+
+	// inject snippets code into content
+	codeSnippets := renderCodeSnippets(snippets)
+	contentBytes := bytes.Replace(all, []byte(marker), codeSnippets, 1)
```

Comment:
```suggestion
	contentBytes := bytes.Replace(all, []byte(marker), codeSnippets, -1)
```
Why do we want to replace at most once?  Sure usually it won't be needed, but I can imagine that it might.  E.g. for debugging I might try to get the string into a `<pre>` block and be able to read what the server generated on the page.